app/services/JourneyProgressService.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.UserAnswers
+import pages.{AddItemPage, CheckAnswersPage, EmptyWaypoints, Page, QuestionPage, Terminus}
+
+class JourneyProgressService {
+
+  def continue(startFrom: Page, answers: UserAnswers): Page = {
+
+    def nextPage(currentPage: Page): Page = currentPage match {
+      case a: AddItemPage with Terminus =>
```

Comment:
Can we rename the single letter variables? Helps when scanning code

terminusAddItemPage
terminus
addItemPage
checkAnswersPage
questionPage
page

shift-f6 rename should make it easy (a lot of things Intellij will auto suggest on rename)